src/ast/printing.jl

```diff
@@ -208,8 +208,8 @@ end
 
 function print_body(io::IO, cmd::NodeCommand, t::Terminal)
     section(io, "Commands")
-    for each in values(cmd.subcmds)
-        print_name_brief(io, each, t)
+    for k in sort(collect(keys(cmd.subcmds)))
```

Comment:
if it's ordered now why do we need to sort it again?

Reply:
maybe we should just sort things in the `cast` macro and store them in an `OrderedDict`? It'd be nice if this PR doesn't affect runtime too much.

Reply:
Thanks for noticing! I got confused about this one, as we can't sort them in `cast`, because top-level commands are passed from `m.CASTED_COMMANDS` [in the runtime](https://github.com/comonicon/Comonicon.jl/blob/master/src/frontend/cast.jl#L432), which has to be a normal `Dict`. However, we can do sorting in the `NodeCommand` [constructor](https://github.com/comonicon/Comonicon.jl/blob/67c7e6068b233ed4f05dfe7089ee776b9a08b5eb/src/ast/types.jl#L169), which I will do.

Reply:
for a project-based CLI, `CASTED_COMMANDS` actually happens in compile time, this is because `@cast` will generate a bunch of code that will be executed during precompile of the package, so in actual runtime (`command_main` generated by code generators) `CASTED_COMMANDS` will not be calculated again. I think it's ok to also sort `CASTED_COMMANDS` inside `@cast`

Reply:
Oh actually I think we don't need to sort CASTED_COMMANDS, the order is stored in each command AST object anyways, otherwise, we will break the promise that the generated code only depends on Julia base